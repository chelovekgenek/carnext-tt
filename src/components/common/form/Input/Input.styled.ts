import styled from "styled-components"

import { FormItem } from "../FormItem"

export const Container = styled(FormItem)`
  display: grid;
`

export const Label = styled.label`
  font-size: 1.2rem;
  font-weight: bold;
`
export const Input = styled.input`
  font-size: 1.2rem;
  padding: 0.2rem;
`
