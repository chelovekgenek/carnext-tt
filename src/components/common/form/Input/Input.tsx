import React from "react"
import { Field, FieldProps } from "formik"

import { FormItemError } from "../FormItem"
import * as Styles from "./Input.styled"

interface IProps {
  name: string
  label: string
}

export const Input: React.FC<IProps> = ({ name, label }) => (
  <Field name={name}>
    {({ field, form: { touched, errors }, ...props }: FieldProps) => (
      <Styles.Container>
        <Styles.Label htmlFor={name}>{label}</Styles.Label>
        <Styles.Input id={name} type="text" value={field.value || ""} {...field} {...props} />
        <FormItemError>{touched[field.name] && errors[field.name]}</FormItemError>
      </Styles.Container>
    )}
  </Field>
)
