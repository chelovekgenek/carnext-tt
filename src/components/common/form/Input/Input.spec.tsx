import React from "react"
import { shallow } from "enzyme"

import fieldProps from "mocks/formik-field-props"

import { InnerFormItem } from "./Input"

describe("Input", () => {
  const defaultProps = {
    ...fieldProps,
    field: { ...fieldProps.field, value: false, name: "input" },
    name: "input",
    label: "input",
  }
  const render = (props: React.ComponentProps<typeof InnerFormItem> = defaultProps) =>
    shallow(<InnerFormItem {...props} />)

  it("should render", () => {
    expect(render()).toMatchSnapshot()
  })
})
