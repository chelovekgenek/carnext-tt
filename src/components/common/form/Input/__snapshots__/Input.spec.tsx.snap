// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render 1`] = `
<Styled(styled.div)>
  <styled.label
    htmlFor="input"
  >
    input
  </styled.label>
  <styled.input
    id="input"
    name="input"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    type="text"
    value={false}
  />
  <styled.div />
</Styled(styled.div)>
`;
